src/main/java/net/minestom/server/component/DataComponentMapImpl.java

```diff
@@ -68,6 +70,26 @@ public boolean has(@NotNull DataComponentMap prototype, @NotNull DataComponent<?
         return new DataComponentMapImpl(newComponents);
     }
 
+    @Override
+    public void forEach(@NotNull BiConsumer<DataComponent<?>, Object> consumer) {
+        // TODO: This only works for itemComponent, fix
+        components.forEach((integer, object) -> {
+            DataComponent<?> component = ItemComponent.fromId(integer);
+            if (component != null) {
+                consumer.accept(component, object);
+            }
+        });
+    }
+
+    @Override
+    public void patchForEach(@NotNull BiConsumer<DataComponent<?>, @Nullable Object> consumer) {
+        // TODO: This only works for itemComponent, fix
```

Comment:
Can you remove or open a issue for the todo ?

Reply:
See https://github.com/Minestom/Minestom/pull/2501#discussion_r1895745862, this won't be merged without some solution being figured out.

Reply:
Then we should consider it a “draft” until then